plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/IGoogleLoginService.java

```diff
@@ -1,50 +1,96 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations under
- * the License.
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  *******************************************************************************/
 
 package com.google.cloud.tools.eclipse.appengine.login;
 
-import com.google.api.client.auth.oauth2.Credential;
+import com.google.cloud.tools.ide.login.Account;
+
+import java.util.Set;
 
 public interface IGoogleLoginService {
 
   /**
-   * Returns the credential of an active user (among multiple logged-in users). A login screen
-   * may be presented, e.g., if no user is logged in or login is required due to an expired
-   * credential. This method returns {@code null} if a user cancels the login process.
-   * For this reason, if {@code null} is returned, the caller should cancel the current
-   * operation and display a general message that login is required but was cancelled or failed.
+   * Initiates user login by launching an external browser that a user will interact with
+   * to log in. The returned {@link Account}, if not {@code null}, becomes an active account.
+   *
+   * Must be called from a UI context.
+   *
+   * @param dialogMessage custom login dialog message. Can be {@code null}
+   * @return signed-in {@link Account} for successful login; {@code null} otherwise,
+   *     including failed and canceled login
+   */
+  Account logIn(String dialogMessage);
+
+  /**
+   * Returns an active {@link Account} (among multiple logged-in accounts). Unlike {@link
+   * #getActiveAccountWithAutoLogin}, this version does not involve login process or make
+   * network calls. Returns {@code null} if there is no account logged in.
+   *
+   * Safe to call from non-UI contexts.
+   */
+  Account getActiveAccount();
+
+  /**
+   * Returns an active {@link Account} (among multiple logged-in accounts). If there is no
+   * account logged in, calls {@link #logIn}. The implementation is effectively as below:
+   *
+   * <pre>
+   * {@code
+   * if (activeAccount != null) {
+   *    return activeAccount;
+   *  }
+   *  return logIn(dialogMessage);
+   * }
+   * </pre>
    *
    * Must be called from a UI context.
    *
-   * @param dialogMessage custom login dialog message. Can be null
+   * @see #logIn
+   */
+  Account getActiveAccountWithAutoLogin(String dialogMessage);
```

Comment:
I'm not a big fan of this name as _auto-login_ doesn't really correspond to the behaviour.  It would be good to use a different prefix for methods that may block.  Maybe `obtain`?  So `getActiveAccount()` will return null if there's no active account, but `obtainActiveAccount()`?